sshd-common/src/main/java/org/apache/sshd/common/future/VerifiableFuture.java

```diff
@@ -35,45 +35,53 @@
     /**
      * Wait {@link Long#MAX_VALUE} msec. and verify that the operation was successful
      *
+     * @param  options     Optional {@link CancelOptions} defining the behavior on time-out or interrupt; ignored if the
+     *                     future is not {@link Cancellable}.
      * @return             The (same) future instance
      * @throws IOException If failed to verify successfully on time
      * @see                #verify(long)
      */
-    default T verify() throws IOException {
-        return verify(Long.MAX_VALUE);
+    default T verify(CancelOption... options) throws IOException {
+        return verify(Long.MAX_VALUE, options);
     }
 
     /**
      * Wait and verify that the operation was successful
      *
      * @param  timeout     The number of time units to wait
      * @param  unit        The wait {@link TimeUnit}
+     * @param  options     Optional {@link CancelOptions} defining the behavior on time-out or interrupt; ignored if the
+     *                     future is not {@link Cancellable}.
      * @return             The (same) future instance
      * @throws IOException If failed to verify successfully on time
      * @see                #verify(long)
      */
-    default T verify(long timeout, TimeUnit unit) throws IOException {
-        return verify(unit.toMillis(timeout));
+    default T verify(long timeout, TimeUnit unit, CancelOption... options) throws IOException {
```

Comment:
May be source-compatible but is not binary-compatible: https://github.com/jenkinsci/bom/pull/2083#issuecomment-1553733703